pyproject.toml

```diff
@@ -0,0 +1,191 @@
+keywords = ["xclim", "xarray", "climate", "climatology", "bias correction", "ensemble", "indicators", "analysis"]
+license = {file = "LICENSE"}
+classifiers = [
+  "Development Status :: 4 - Beta",
+  "Intended Audience :: Developers",
+  "Intended Audience :: Science/Research",
+  "License :: OSI Approved :: Apache Software License",
+  "Natural Language :: English",
+  "Operating System :: OS Independent",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Topic :: Scientific/Engineering :: Atmospheric Science"
+]
+dynamic = ["description", "version"]
+dependencies = [
+  "boltons>=20.1",
+  "bottleneck>=1.3.1",
+  "cf-xarray>=0.6.1",
+  "cftime>=1.4.1",
+  "Click>=8.1",
+  "dask[array]>=2.6",
+  "jsonpickle",
+  "numba",
+  "numpy>=1.16",
+  "pandas>=0.23",
+  "pint>=0.10",
+  "pyyaml",
+  "scikit-learn>=0.21.3",
+  "scipy>=1.2,<1.9", # see: https://github.com/Ouranosinc/xclim/issues/1142
+  "statsmodels",
+  "xarray>=2022.06.0"
+]
+
+[project.optional-dependencies]
+dev = [
+  # Dev tools and testing
+  "black >=22.12",
+  "blackdoc",
+  "bump2version",
+  "coverage",
+  "flake8",
+  "flake8-rst-docstrings",
+  "ipython",
+  "mypy",
+  "nbqa",
+  "nbval",
+  "pre-commit >=2.9",
+  "pydocstyle >=5.1.1",
+  "pybtex",
+  "pylint <2.15",
+  "pytest",
+  "pytest-cov",
+  "pytest-xdist[psutil]",
+  "tokenize-rt",
+  "tox",
+  "wheel",
+  "xdoctest",
+  "yamale",
+  "yamllint",
+  # Documentation and examples
+  "distributed >=2.0",
+  "ipykernel",
+  "matplotlib",
+  "nbsphinx",
+  "nc-time-axis",
+  "netCDF4 <1.6",
+  "pooch",
+  "sphinx",
+  "sphinx-codeautolink",
+  "sphinx-copybutton",
+  "sphinx-rtd-theme >=1.0",
+  "sphinxcontrib-bibtex"
+]
+
+[project.scripts]
+xclim = "xclim.cli:cli"
+
+[project.urls]
+"Homepage" = "https://xclim.readthedocs.io/"
+"Source" = "https://github.com/Ouranosinc/xclim/"
+"Changelog" = "https://xclim.readthedocs.io/en/stable/history.html"
+"Issue tracker" = "https://github.com/Ouranosinc/xclim/issues"
+"About Ouranos" = "https://www.ouranos.ca/en/"
+
+[tool]
+
+[tool.black]
+target-version = [
+  "py38"
+]
+
+[tool.coverage.run]
+relative_files = true
+omit = ["*/tests/*.py"]
+
+[tool.flit.sdist]
+include = ["AUTHORS.rst", "CONTRIBUTING.rst", "HISTORY.rst", "LICENSE", "README.rst", "requirements_upstream.txt", "xclim/**/*.py", "xclim/**/*.json", "xclim/**/*.yml", "xclim/**/*.txt", "docs/**/*.rst", "docs/conf.py", "docs/Makefile", "docs/make.bat", "docs/**/*.jpg", "docs/**/*.png", "docs/**/*.gif"]
+exclude = [".*", "pylintrc", "**/__pycache__", "**/*.py[co]", "docs/_*", "docs/xclim*.rst", "docs/modules.rst"]
+
+[tool.isort]
+profile = "black"
+py_version = 38
+append_only = true
+add_imports = "from __future__ import annotations"
+
+[tool.mypy]
+python_version = 3.8
+show_error_codes = true
+warn_return_any = true
+warn_unused_configs = true
+
+[[tool.mypy.overrides]]
```

Comment:
According to [mipy doc  ](https://mypy.readthedocs.io/en/stable/config_file.html#example-pyproject-toml), you can group the `ignore_missing_imports` in a single `tool.mypy.overrides` entry: 
```
[[tool.mypy.overrides]]
module = [
    "xarray.*",
    "numpy.*"
]
ignore_missing_imports = true
```

Reply:
Right. I read somewhere about how splitting them up is preferable? At this point in time, we aren't using mypy, so let's make this change.